### PR TITLE
Fix field splittings by missing double quotations

### PR DESCRIPTION
**Pull request by a5ob7r**
Expansions with unintentional field splitting by shell maybe cause bugs, so we usually wrap them with double quotes to prevent the bugs.

In practice, PaSH has an inconsistent argument handling bug because the current implementation wraps the potential parameter expansion with double quotes at one place but doesn't at another one. For example, PaSH raises a parsing error when running the command below.

```sh
$ pa.sh -c 'cat foo bar | tr -dc abc'
...
pash_runtime_daemon.py: error: argument -d/--debug: invalid int value: 'c'
...
```

To fix this error and prevent similar potential bugs, wrap expansions with double quotes and test using shellcheck to check missing quotations automatically.

### CODE REVIEW

**Comment by angelhof** [comment]
Thanks a lot for this PR and for finding this bug! The changes look good to me but it would be good to get reviews from the rest of the core team since the changes span multiple files :)

File: .github/workflows/tests.yaml
Comment (angelhof) [inline .github/workflows/tests.yaml:91]: We probably also need to ignore everything in `evaluation` and all other auxiliary directories.

Comment (a5ob7r) [reply .github/workflows/tests.yaml:91]: OK. I'll add the path to `ignore_paths` after merging https://github.com/a5ob7r/pash/pull/1. In addition, we should add the `scripts` dir to it too?

Comment (angelhof) [reply .github/workflows/tests.yaml:91]: I looked at the ones that fail, and I would suggest adding all of them (`annotations`, `evaluation`, `runtime`, `scripts`). 

We could do another PR on top of this branch of yours so that we check that shellcheck CI passes before we actually merge it here (so that we don't introduce a failing CI workflow).

File: runtime/split_pipe.sh
Comment (mgree) [inline runtime/split_pipe.sh:1]: Whoa... how did _this_ ever work?

**Review by mgree — APPROVED** [review]
Looks good to me. It might be good to add some valid tests that both do and _don't_ quote, to make sure quoting doesn't become load bearing.

File: runtime/split_pipe.sh
Comment (angelhof) [reply runtime/split_pipe.sh:1]: Unclear if this is used :')

**Review by nvasilakis — APPROVED** [review]
This looks great, thank you!

File: compiler/pash_runtime.sh
Comment (angelhof) [inline compiler/pash_runtime.sh:143]: This is a mistake. We want this to be a bash array and so we need to preserve the spaces. @a5ob7r could you revert this? This is the reason why the CI fails.

**Review by angelhof — CHANGES_REQUESTED** [review]
The changes broke pash when we wanted to create a bash array with multiple entries. Could you revert the change that I commented on?

File: compiler/test_evaluation_scripts.sh
Comment (angelhof) [inline compiler/test_evaluation_scripts.sh:188]: The current problem is somewhere here, though I am not sure where and why...

Comment (angelhof) [reply compiler/test_evaluation_scripts.sh:188]: Well, I think I found the problem. We need to remove the quotes from `$assert_correctness` and `$conf` because these contain spaces on purpose.

File: compiler/test_evaluation_scripts.sh
Comment (angelhof) [inline compiler/test_evaluation_scripts.sh:156]: This one needs to be reverted too, since it also contains spaces on purpose.

**Comment by a5ob7r** [comment]
Or we should revert the change of `compiler/test_evaluation_scripts.sh` and add it to `ignore_paths` to skip checking it? Come to think of it, this file is for tests, not for runtime.

**Review by angelhof — APPROVED** [review]

**Pull request merged**
